### PR TITLE
[GCS] Add WorkerInfoAccessor to GCS Client

### PR DESCRIPTION
**Pull request by micafan**
## Why are these changes needed?

Add WorkerInfoAccessor to GCS Client.

## Related issue number
Closes #5058

## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [X] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: src/ray/gcs/accessor.h [hunk 1]
```diff
@@ -481,6 +481,11 @@ class StatsInfoAccessor {
  public:
   virtual ~StatsInfoAccessor() = default;
 
+  /// Add profile data to GCS asynchronously.
+  ///
+  /// \param data_ptr The profile data that will be added to GCS.
+  /// \param callback Callback that will be called when add is complete.
```
Comment (ffbin) [inline src/ray/gcs/accessor.h]: completed

**Comment by ffbin** [comment]
LGTM

**Review by ffbin — APPROVED** [review]

File: src/ray/gcs/accessor.h (hunk 1 above)
Comment (micafan) [reply src/ray/gcs/accessor.h]: complete

**Comment by zhijunfu** [comment]
@micafan There are some conflicts, could you merge latest master?

**Review by zhijunfu — APPROVED** [review]

**Comment by micafan** [comment]
@stephanie-wang Could you take a look at this PR?

**Comment by stephanie-wang** [comment]
Sorry for the late reply, but it looks good!

**Pull request merged**